sdks/go/pkg/beam/core/metrics/metrics.go

```diff
@@ -754,6 +770,19 @@ func (r MsecResult) Result() MsecValue {
 	return r.Attempted
 }
 
+// Name returns the Name of this Gauge.
+func (r MsecResult) Name() string {
+	return r.Key.Name
+}
+
+// Namespace returns the Namespace of this Gauge.
+func (r MsecResult) Namespace() string {
+	return r.Key.Namespace
+}
```

Comment:
Msec results can't have any name or namespaces (or at the very least, won't have them correctly populated here).  return `""` instead for now.

Also, please fix the copypasta "Gauge" references.